R/utils.R

```diff
@@ -125,7 +123,7 @@ generate_dust_support_sum <- function(rank) {
 dust_type <- function(type) {
   switch(type,
          double = "real_t",
-         int = "int_t",
+         int = "int",
```

Comment:
Just confirming that `int` is the correct type? I am bad at remembering exactly how this works, but iirc this could change based on compiler/platform. Is there any advantage in using `int32_t` or `long int` which are always at least 32 bits?

Reading here: https://en.cppreference.com/w/cpp/language/types

Reply:
My understanding reading this is that on any 64 bit system, `int` will be at least 32 bits

Reply:
R checks for int being a sensible length. If we add this back in, it needs to be done in dust too, but is not (not done there because we had no usecase and then no tests so I removed it!)

Reply:
https://cran.r-project.org/doc/manuals/R-exts.html#Interface-functions-_002eC-and-_002eFortran

> On all R platforms int and INTEGER are 32-bit. Code ported from S-PLUS (which uses long * for logical and integer) will not work on all 64-bit platforms (although it may appear to work on some, including Windows). Note also that if your compiled code is a mixture of C functions and Fortran subprograms the argument types must match as given in the table above.